From 444718caad4fa7f37fb50c6c46566e5d27ad5df6 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Wed, 25 Apr 2018 03:03:03 +0200
Subject: [PATCH 2/2] hal: msim_voice_extn: Set msim_phone based on phone_type
 parameter

* This is consistently cp1 on SIM1 and cp2 on SIM2, so drop all the
  confusing and unnecessary property_get calls, and just use the param

Change-Id: I4a59ebd177d46b47e00067af7e6dce49d3e7e676
---
 hal/voice_extn/msim_voice_extn.c | 24 +-----------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

diff --git a/hal/voice_extn/msim_voice_extn.c b/hal/voice_extn/msim_voice_extn.c
index 199dd7c75..90ec47ed0 100644
--- a/hal/voice_extn/msim_voice_extn.c
+++ b/hal/voice_extn/msim_voice_extn.c
@@ -34,14 +34,6 @@
 #define AUDIO_PARAMETER_VALUE_CP1 "cp1"
 #define AUDIO_PARAMETER_VALUE_CP2 "cp2"
 
-#ifdef HTC_DUAL_SIM
-#define RADIO_PREFER_NETWORK_SLOT0 "persist.radio.prefer.network"
-#define RADIO_PREFER_NETWORK_SLOT1 "persist.radio.prefer.nw.sub"
-#elif SAMSUNG_DUAL_SIM
-#define AUDIO_PROPERTY_SEC_VSID1 "gsm.current.vsid"
-#define AUDIO_PROPERTY_SEC_VSID2 "gsm.current.vsid2"
-#endif
-
 #define VOICE2_VSID 0x10DC1000
 
 #define VOICE2_SESS_IDX (VOICE_SESS_IDX + 1)
@@ -93,21 +85,7 @@ int msim_voice_extn_set_parameters(struct audio_device *adev __unused,
     ret = str_parms_get_str(parms, AUDIO_PARAMETER_KEY_PHONETYPE, value,
                             sizeof(value));
     if (ret >= 0) {
-#ifdef HTC_DUAL_SIM
-        voice_slot = property_get_int32(RADIO_PREFER_NETWORK_SLOT0) != 1 ? 0 : 1;
-        if (property_get_int32(RADIO_PREFER_NETWORK_SLOT0) ==
-                property_get_int32(RADIO_PREFER_NETWORK_SLOT1)) {
-            voice_slot = 0;
-        }
-        if (!strcmp(value, AUDIO_PARAMETER_VALUE_CP2)) {
-            msim_phone = voice_slot == 0 ? MSIM_SIM2 : MSIM_SIM1;
-        } else {
-            msim_phone = voice_slot == 0 ? MSIM_SIM1 : MSIM_SIM2;
-        }
-#elif SAMSUNG_DUAL_SIM
-        msim_phone = property_get_int32(!strcmp(value, AUDIO_PARAMETER_VALUE_CP2) ?
-                AUDIO_PROPERTY_SEC_VSID2 : AUDIO_PROPERTY_SEC_VSID1) != 0 ? MSIM_SIM2 : MSIM_SIM1;
-#endif
+        msim_phone = !strcmp(value, AUDIO_PARAMETER_VALUE_CP2) ? MSIM_SIM2 : MSIM_SIM1;
         ALOGV("%s: phone_type: %d", __func__, msim_phone);
     }
 
-- 
2.17.0

